### PR TITLE
Switch to new version of install-flox-action 

### PR DESCRIPTION
**Pull request by garbas**
And also:
- run again, since it was removed from CI, `cargo tests`
- we run Cargo tests in parallel to Bats tests
- sneaking in support for direnv (`.envrc` file)

### CODE REVIEW

**Comment by stahnma** [comment]
Can this please:
1. Get attached to an issue.
2. Get an estimate.

**Comment by garbas** [comment]
> Can this please:
> 
>     1. Get attached to an issue.
> 
>     2. Get an estimate.

I need an emoji that says: _sorry, I knew what I had to do, but failed/forgot, will do better next time_

**Comment by garbas** [comment]
I've added the ticket. I need at least one green 👍 to merge this.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,132 +13,102 @@ concurrency:
...
+        restore-keys: cargo-${{ runner.os }}-
+
+    - name: "Cargo Test"
       run: |
-        nix build '.#flox-bash' -L --no-update-lock-file      \
-                                --print-out-paths --no-link;
+        nix develop -L \
+          --extra-experimental-features "nix-command flakes" \
+          --no-update-lock-file \
+          '.#flox' \
```
Comment (ysndr) [inline .github/workflows/ci.yml:59]: TODO: change to .#ci

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -13,132 +13,102 @@ concurrency:
...
+            --command cargo test --locked --workspace;
```
Comment (ysndr) [inline .github/workflows/ci.yml:60]: use frozen (vendored) deps `flox`

```
nix build .\#flox.cargoVendorDir --out-link .cargo
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -13,132 +13,102 @@ concurrency:
...
-          wc -l "$NEW_STORE_PATHS_FILE"|cut -d' ' -f1;
-        ) new $NIX_STORE_DIR paths" >&2;
-        # Allow pushing to fail.
-        cat "$NEW_STORE_PATHS_FILE"                            \
-          |xargs -r nix copy --to "$FLOX_SUBSTITUTER" -vv||:;
-
-    # These cannot be run offline so we need to build in our tree.
-    - name: Rust Tests
+        nix build '.#flox' -L \
+            --extra-experimental-features "nix-command flakes" \
```
Comment (ysndr) [inline .github/workflows/ci.yml:91]: install-flox-action installs nix as upstream without extra configuration.
Product decision post GA

ARMADILLO

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -13,132 +13,102 @@ concurrency:
...
-    - name: Build flox-tests
-      run: |
-        nix build '.#flox-tests' -L --no-update-lock-file      \
-                                 --print-out-paths --no-link;
+    - name: "Install flox"
+      uses: flox/install-flox-action@main
+      with:
+        github-access-token: ${{ secrets.NIX_GIT_TOKEN }}
+        substituter: s3://flox-store
+        substituter-key: ${{ secrets.FLOX_STORE_PUBLIC_NIX_SECRET_KEY }}
```
Comment (ysndr) [inline .github/workflows/ci.yml:84]: 🤔

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -13,132 +13,102 @@ concurrency:
...
 
-    - name: Build flox
-      run: |
-        nix build '.#flox' -L --no-update-lock-file --print-out-paths;
-        echo "FLOX_CLI=$( readlink -f ./result; )/bin/flox" >> "$GITHUB_ENV";
-        rm ./result;
+    steps:
+    - name: "Checkout"
+      uses: actions/checkout@v3
+      with:
```
Comment (ysndr) [inline .github/workflows/ci.yml:76]: consider removing this for "faster" clones.
previously needed to facilitate `publish`

**Review by ysndr — APPROVED** [review]

**Pull request merged**